lib/graphql_schema.rb

```diff
@@ -2,6 +2,24 @@
 require 'set'
 
 class GraphQLSchema
+  INVALID_SCHEMA = Class.new(StandardError)
+
+  class << self
+    def load_schema(schema)
+      case schema
+      when Hash
+        new(schema)
+      when GraphQLSchema
+        schema
+      when Pathname, String
+        schema = File.read(schema) if Pathname(schema).file?
```

Comment:
Does this mean that if it is called with a JSON string then it will first treat the string as a filename and look for a file with that filename?  I would prefer to just have separate `GraphQLSchema.parse(json_string)` and `GraphQLSchema.read(filename)` methods, which would simplify the code and make the intent of the code at the call-site clearer.